File: bin/sp-4.4.1/sicstus-4.4.1/se/sics/prologbeans/PBNil.java
/**
 * Copyright (c) 2003 SICS AB. All rights reserved.
 */
package se.sics.prologbeans;

/**
 * <code>PBNil</code> is the representation of the empty list, [].
 */

class PBNil extends PBAtom
{
    /**
     * Creates a new <code>PBNil</code> instance with the specified name.
     */
    PBNil()
    {
        super("[]");
    }

    public boolean isEmptyList()
    {
        return true;
    }

    public boolean isProperList()
    {
        return true;
    }

    public int length()
    {
        return 0;
    }

    public String getString()
    {
        return "";
    }
}

File: bin/sp-4.4.1/sicstus-4.4.1/library/prologbeans.NET/PBNil.cs

namespace se.sics.prologbeans
{
	
	/// <summary> <code>PBNil</code> is the representation of the empty list, [].</summary>
	
	class PBNil:PBAtom
	{
		override public bool EmptyList {
			get {
				return true;
			}
			
		}

		override public bool ProperList {
			get {
				return true;
			}
			
		}

		/// <summary> Creates a new <code>PBNil</code> instance with the specified name.</summary>
		internal PBNil () : base ("[]")
		{
		}

		public override int length ()
		{
			return 0;
		}

		public override string getString ()
		{
			return "";
		}
	}
}